plugins/modules/database/saphana/hana_query.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2021, Rainer Leber <rainerleber@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: hana_query
+
+short_description: Execute SQL on HANA.
+
+version_added: "3.2.0"
+
+description: This module executes SQL statements on HANA with hdbsql.
+
+options:
+    sid:
+        description: The system ID.
+        type: str
+        required: true
+    instance:
+        description: The instance number.
+        type: str
+        required: true
+    user:
+        description: A dedicated username. Defaults to C(SYSTEM).
+        type: str
+        default: SYSTEM
+    password:
+        description: The password to connect to the database.
+        type: str
+        required: true
+    autocommit:
+        description: Autocommit the statement.
+        type: bool
+        default: True
+    host:
+        description: The Host IP address. The port can be defined as well.
+        type: str
+    database:
+        description: Define the database on which to connect.
+        type: str
+    encrypted:
+        description: Use encrypted connection. Defaults to C(false).
+        type: bool
+        default: false
+    filepath:
+        description:
+        - One or more files each containing one SQL query to run.
+        - Must be a string or list containing strings.
+        type: list
+        elements: path
+    query:
+        description:
+        - SQL query to run. Multiple queries can be passed using YAML list syntax.
+        - Must be a string or list containing strings.
+        type: list
+        elements: str
+
+author:
+    - Rainer Leber (@rainerleber)
+'''
+
+EXAMPLES = r'''
+
+- name: Simple select query
+  community.general.hana_query:
+    sid: "hdb"
+    instance: "01"
+    password: "Test123"
+    query: "select user_name from users"
+
+- name: Run several queries
+  community.general.hana_query:
+    sid: "hdb"
+    instance: "01"
+    password: "Test123"
+    query:
+    - "select user_name from users;"
+    - select * from SYSTEM;
+    host: "localhost"
+    autocommit: False
+
+- name: Run several queries from file
+  community.general.hana_query:
+    sid: "hdb"
+    instance: "01"
+    password: "Test123"
+    filepath:
+    - /tmp/HANA_CPU_UtilizationPerCore_2.00.020+.txt
+    - /tmp/HANA.txt
+    host: "localhost"
+
+
+'''
+
+import io
+import csv
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+def csv_to_json(rawcsv):
+    lines = rawcsv[:rawcsv.rfind('\n')]
+    reader_raw = csv.DictReader(io.StringIO(lines))
+    reader = [dict((k, v.strip()) for k, v in row.items()) for row in reader_raw]
+    tolist = list(reader)
+    return json.dumps(tolist)
```

Comment:
I don't think it makes sense to return encoded JSON. Why not return real JSON so the module's result is simple to use?

Reply:
I didn't know exactly what you mean the returned value is like this:
```ok: [10.2.0.16] => {
    "consistency_exec.stdout": [
        {
            "AVG_TIME_S": "0.65",
            "CHECK_ACTION": "CHECK",
            "CHECK_PROCEDURE_NAME": "CHECK_TABLE_CONSISTENCY",
            "DB_USER": "SYSTEM",
            "ERROR_DETAILS": "0",
            "EXECUTIONS": "6",
            "LAST_START_TIME": "2021/05/23 09:11:15",
            "SCHEMA_NAME": "",
            "TABLE_NAME": "",
            "TOT_TIME_H": "0.00"
        }
    ]
}```

Reply:
That looks like the return value already got processed. The module is returning a string with JSON in it, not proper JSON. You can run your playbook with `-vvv` to see what the module actually returns.

Also you should add this to your tests, for example I would expect that `SELECT * FROM users:` returns something else than an empty string :)

Reply:
make sense :-) I will change it to a better format

Reply:
Making `out` a list where every entry corresponds to one query made is probably the best way to do this. The result variable should probably not be called `out` though :)